app/services/vacancy_facets.rb

```diff
@@ -0,0 +1,66 @@
+class VacancyFacets
+  FIELDS = %i[job_roles subjects cities counties].freeze
+
+  def initialize(store: Redis.new)
+    @store = store
+  end
+
+  def get(field)
+    return {} if !FIELDS.include?(field) || !store.exists?(field)
+
+    JSON.parse(store.get(field))
+  end
+
+  def refresh
+    set_job_role_facets
+    set_subjects_facets
+    set_cities_facets
+    set_counties_facets
+  end
+
+private
+
+  attr_reader :store
+
+  def set_job_role_facets
+    store.set(:job_roles, job_role_facet(Vacancy::JOB_ROLE_OPTIONS).to_json)
+  end
+
+  def set_subjects_facets
+    store.set(:subjects, subject_facet(SUBJECT_OPTIONS).to_json)
+  end
+
+  def set_cities_facets
+    store.set(:cities, location_category_facet(CITIES).to_json)
+  end
+
+  def set_counties_facets
+    store.set(:counties, location_category_facet(COUNTIES).to_json)
+  end
+
+  def job_role_facet(job_roles)
+    job_roles.filter_map { |job_role|
```

Comment:
Sometimes I find it clearer to explicitly include the unused parameter like `|job_role, _|` when it's a hash but we only care about one of the key/value